### DISCUSSION TITLE
Looking for Attestation Contract ABI.

### DISCUSSION THREAD

**Discussion by realchoubey** [celo-org/social-connect#220] [Help (Q&A)] [answered]
I'm looking to index attestation contract event data:
https://explorer.celo.org/address/0xD787384d91a7fFaC85d51C63EF71580df6C677B7/contracts

https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/identity/Attestations.sol
Can someone point me to the attestation contracts ABI, because implementation contract is not verified.

cc: @0xarthurxyz

**Comment by critesjosh** [comment, marked as answer]
It looks like this page has the Attestation.sol ABI https://explorer.celo.org/address/0x936d74304a17440488e02197285a2EA5f6Dd40A8/contracts

**Comment by carterqw2** [comment]
Here is the latest implementation verified on Sourcify 
https://explorer.celo.org/address/0x1EC3366D384ee7996F2F70B67A65C5d54Ce96040/contracts
https://repo.sourcify.dev/contracts/partial_match/42220/0x1EC3366D384ee7996F2F70B67A65C5d54Ce96040/

**Comment by arthurgousset** [comment]
Thanks @critesjosh and @carterqw2 for answering so quickly! 🙏

**Comment by realchoubey** [comment]
@carterqw2 @critesjosh 

1. What is the difference between below three Attestation contracts:
https://explorer.celo.org/address/0x936d74304a17440488e02197285a2EA5f6Dd40A8/contracts
https://repo.sourcify.dev/contracts/partial_match/42220/0x1EC3366D384ee7996F2F70B67A65C5d54Ce96040/
https://explorer.celo.org/address/0xD787384d91a7fFaC85d51C63EF71580df6C677B7/transactions

2. Which one to use for indexing Attestation transactions ?
3. As per events we don't see direct phone number, can you please let me know which event parameter is phone number.
https://github.com/celo-org/celo-monorepo/blob/83b5c21c20fb22d17c46f73bc67b3de88f69e171/packages/protocol/contracts/identity/Attestations.sol#L97

**Reply by critesjosh** [reply]
1. I am not exactly sure about the differences between the first two, but the third contract is a proxy contract that allows the Attestations contract to be upgraded. You can read more about the proxy upgrade pattern [here](https://docs.openzeppelin.com/upgrades-plugins/1.x/proxies).
2. I think you'd want to watch [this contract](https://explorer.celo.org/address/0xdC553892cdeeeD9f575aa0FBA099e5847fd88D20/logs) (another attestations proxy). If you look at the logs it includes events like `AttestationCompleted`.
3. The contract doesn't log the user's phone number--the phone number is considered private data. A phone hash is used as the identifier, `hash(phoneNumber + pepper)`. The pepper is queryable via ODIS. You can find an example of how to do that in javascript [here](https://github.com/critesjosh/register-number).

**Reply by realchoubey** [reply]
Thanks @critesjosh. I am from Dapplooker and implementing subgraph/dashboard for attestation contract transactions. This is very helpful

**Reply by eelanagaraj** [reply]
Agreed with @critesjosh! I believe you want the `AttestationsProxy` linked in (2).
You can also check the current implementation and proxy addresses for the core Celo contracts on `mainnet` via `celocli network:contracts --node https://forno.celo.org/`. (For alfajores, just use `--node https://alfajores-forno.celo-testnet.org` or another alfajores node!)

**Reply by realchoubey** [reply]
@carterqw2 @critesjosh @eelanagaraj 
Thanks for your helps guys! I'm able to deploy the sub-graph.

It would be great if you guys can give me some idea on what will be useful in terms of Analytics of Attestation?

**Reply by arthurgousset** [reply]
Glad you found your answers @realchoubey! 
Please could you take the analytics questions for the attestations contract into a separate discussion topic.
This way, we keep the threads clean and easily searchable. Thanks 🙏

**Comment by arthurgousset** [comment]
For future reference, here is also a useful page with proxy and implementation contract addresses for the "core contracts": https://dhutch-test.celo-networks-dev.org/

<img width="1421" alt="image" src="https://user-images.githubusercontent.com/46296830/161323831-1d3f1cc3-bdde-40ca-aba6-5972dc1b255a.png">

**Comment by arthurgousset** [comment]
Quick ping because I transferred this discussion 

- from: `celo-org/celo-monorepo/discussions`
- to: `celo-org/identity/discussions`

to be more flexible in the types of discussions we have:

- 💡 [Feature requests (Feedback)](https://github.com/celo-org/identity/discussions/categories/feature-requests-feedback)
- 😺 [Help needed (Q&A)](https://github.com/celo-org/identity/discussions/categories/help-needed-q-a)
- 🗳️ [Polls (Voting)](https://github.com/celo-org/identity/discussions/categories/polls-voting)
- 🙌 [Show and tell (Updates)](https://github.com/celo-org/identity/discussions/categories/show-and-tell-updates)
 
Sorry for any inconvenience 🙏
